prov/rxm/src/rxm_conn.c

```diff
@@ -491,6 +489,17 @@ void rxm_process_connect(struct rxm_eq_cm_entry *cm_entry)
 						    server_conn_id);
 		conn->remote_pid = rxm_peer_pid(cm_entry->data.accept.
 						server_conn_id);
+		conn->peer_flow_ctrl = cm_entry->data.accept.flow_ctrl;
+	}
+
+	if (conn->flow_ctrl & conn->peer_flow_ctrl) {
+		domain = container_of(conn->ep->util_ep.domain,
+				      struct rxm_domain, util_domain);
+		if (!domain->flow_ctrl_ops->enable(conn->msg_ep)) {
+			domain->flow_ctrl_ops->
+				set_threshold(conn->msg_ep,
+					      conn->ep->msg_info->rx_attr->size / 2);
```

Comment:
Should enable and set_threshold be combined into a single call?

Reply:
I think so, given that the available() method already serves the original purpose of status checking.